autoscale-workload-rabbit/src/main/java/com/hpe/caf/autoscale/workload/rabbit/RabbitSystemResourceMonitor.java

```diff
@@ -0,0 +1,90 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.autoscale.workload.rabbit;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hpe.caf.api.autoscale.ScalerException;
+import com.squareup.okhttp.OkHttpClient;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.concurrent.TimeUnit;
+import retrofit.ErrorHandler;
+import retrofit.RestAdapter;
+import retrofit.RetrofitError;
+import retrofit.client.OkClient;
+import retrofit.client.Response;
+import retrofit.http.GET;
+import retrofit.http.Path;
+
+public class RabbitSystemResourceMonitor
+{
+    private final RabbitManagementApi rabbitApi;
+    private final String rabbitEnpoint;
+    private final ObjectMapper mapper = new ObjectMapper();
+    private static final int RMQ_TIMEOUT = 10;
+
+    public RabbitSystemResourceMonitor(final String endpoint, final String user, final String pass)
+    {
+        this.rabbitEnpoint = endpoint;
+        final String credentials = user + ":" + pass;
+        final OkHttpClient ok = new OkHttpClient();
+        ok.setReadTimeout(RMQ_TIMEOUT, TimeUnit.SECONDS);
+        ok.setConnectTimeout(RMQ_TIMEOUT, TimeUnit.SECONDS);
+        // build up a RestAdapter that will automatically handle authentication for us
+        final RestAdapter.Builder builder = new RestAdapter.Builder().setEndpoint(endpoint).setClient(new OkClient(ok));
+        builder.setRequestInterceptor(requestFacade -> {
+            final String str = "Basic " + Base64.getEncoder().encodeToString(credentials.getBytes(StandardCharsets.UTF_8));
+            requestFacade.addHeader("Accept", "application/json");
+            requestFacade.addHeader("Authorization", str);
+        });
+        builder.setErrorHandler(new RabbitApiErrorHandler());
+        final RestAdapter adapter = builder.build();
+        rabbitApi = adapter.create(RabbitManagementApi.class);
+    }
+    
+    public double getCurrentMemoryComsumption() throws ScalerException{
+        
+        try {
+            String rabbitHost = rabbitEnpoint.replace("http://", "");
+            rabbitHost = rabbitHost.substring(0, rabbitHost.indexOf(":"));
+            final Response response = rabbitApi.getNodeStatus(rabbitHost);
+            final JsonNode root = mapper.readTree(response.getBody().in());
+            final long memory_limit = root.get("mem_limit").asLong();
+            final long memory_used = root.get("mem_used").asLong();
+            return ((double)memory_used / memory_limit) * 100;
+        } catch (final IOException ex) {
+            throw new ScalerException("Unable to map response to status request.", ex);
```

Comment:
What a pity to have lost the genuine nature of the exception.  Also, what does that message mean?